src/request/sign-btc-transaction/SignBtcTransaction.js

```diff
@@ -216,43 +216,82 @@ class SignBtcTransaction {
         // Sort outputs by value ASC, then address ASC
         outputs.sort((a, b) => (a.value - b.value) || (a.address < b.address ? -1 : 1));
 
-        try {
-            // Construct transaction
-            const psbt = new BitcoinJS.Psbt({ network: BitcoinUtils.Network });
-
-            // Add inputs
-            // @ts-ignore Argument of type 'Uint8Array' is not assignable to parameter of type 'Buffer'.
-            psbt.addInputs(request.inputs);
-            // Add outputs
-            psbt.addOutputs(outputs);
-
-            // Sign
-            const paths = request.inputs.map(input => input.keyPath);
-            btcKey.sign(paths, psbt);
-
-            // Verify that all inputs are signed
-            if (!psbt.validateSignaturesOfAllInputs()) {
-                throw new Error('Invalid or missing signature(s).');
-            }
 
-            // Finalize
-            psbt.finalizeAllInputs();
+        // Construct transaction
+        const psbt = new BitcoinJS.Psbt({ network: BitcoinUtils.Network });
 
-            // Extract tx
-            const tx = psbt.extractTransaction();
+        // Add inputs
+        // @ts-ignore Argument of type 'Uint8Array' is not assignable to parameter of type 'Buffer'.
+        psbt.addInputs(request.inputs);
+        // Add outputs
+        psbt.addOutputs(outputs);
 
-            /** @type {KeyguardRequest.SignedBitcoinTransaction} */
-            const result = {
-                transactionHash: tx.getId(),
-                raw: tx.toHex(),
-            };
-            resolve(result);
-        } catch (error) {
-            TopLevelApi.setLoading(false);
-            console.error(error);
-            alert(`ERROR: ${error.message}`);
-            // return;
+        if (request.inputs.length === 1 && request.inputs[0].type === 'htlc-refund') {
+            psbt.locktime = HtlcUtils.decodeBtcHtlcScript(request.inputs[0].witnessScript).timeoutTimestamp + 1;
+            psbt.setInputSequence(0, 0xfffffffe); // Signal to use locktime, but do not opt into replace-by-fee
+        }
+
+        // Sign
+        const paths = request.inputs.map(input => input.keyPath);
+        btcKey.sign(paths, psbt);
+
+        // Verify that all inputs are signed
+        if (!psbt.validateSignaturesOfAllInputs()) {
+            throw new Error('Invalid or missing signature(s) for BTC transaction.');
         }
+
+        // Finalize
+        let i = 0;
+        for (const parsedInput of request.inputs) {
+            if (parsedInput.type === 'default') {
+                psbt.finalizeInput(i);
+            }
+            if (parsedInput.type === 'htlc-redeem' || parsedInput.type === 'htlc-refund') {
```

Comment:
The type `htlc-redeem` is currently not used with the standard `SignBtcTransaction` request, but I though it can't hurt to include it, as the code is pretty much the same as for `htlc-refund`, except 2 lines in the middle.